src/runtime/rpc/network.cpp

```diff
@@ -74,9 +74,7 @@ void rpc_session::set_connected()
 
     {
         utils::auto_lock<utils::ex_lock_nr> l(_lock);
-        dassert((_connect_state == SS_NEGOTIATING && security::FLAGS_enable_auth) ||
-                    (_connect_state == SS_CONNECTING && !security::FLAGS_enable_auth),
-                "wrong session state");
+        dassert(_connect_state == SS_CONNECTING, "wrong session state");
```

Comment:
Use `dcheck_eq` then you can get exact value of `_connect_state ` when check failed.